contracts/RecruitmentV2.sol

```diff
@@ -0,0 +1,434 @@
+
+        require(msg.sender == job.company, "Only the company can hire");
+        require(!referral.isHired, "Candidate already hired");
+
+        referral.isHired = true;
+        job.isOpen = false;
+        job.hiredReferralId = referralId;
+        companyHiredCandidate[msg.sender][referral.candidateEmail] = true;
+    }
+
+    /// returns the jobs created by a company
+    /// @param _company address of the company to retrive the jobs
+    function getCompanyJobs(
+        address _company
+    ) public view returns (uint256[] memory) {
+        require(
+            companies[_company].companyAddress != address(0),
+            "Company is not registered"
+        );
+        return companyToJobs[_company];
+    }
+
+    /// returns the referrals made by a referrer
+    /// @param referrer address of the referrer
+    function getReferrerReferrals(
+        address referrer
+    ) public view returns (uint256[] memory) {
+        require(
+            referrers[referrer].reffererAddress != address(0),
+            "Referrer is not registered"
+        );
+        return referrerToReferrals[referrer];
+    }
+
+    /// Close a Job without hiring
+    /// @param jobId job id of the job
+    function closeJob(
+        uint256 jobId
+    ) external onlyRegisteredCompany onlyOpenJob(jobId) {
+        Job storage job = jobs[jobId];
+        require(
+            msg.sender == job.company,
+            "Only the company can close the job"
+        );
+        job.isOpen = false;
+        token.transfer(job.company, job.bounty); // transfer the bounty back to the company
+    }
+
+    /// Dirburse Job bounty after 90 days
+    /// @param _jobId job id of the job for which the bounty is to be disbursed
+    function disburseBounty(
+        uint256 _jobId
+    ) external nonReentrant onlyRegisteredCompany {
+        require(
+            jobs[_jobId].company == msg.sender,
+            "Only the company can disburse bounty"
+        );
+        require(jobs[_jobId].isOpen == false, "Job is still open");
+        require(jobs[_jobId].isDisbursed == false, "Bounty already disbursed");
+        require(
+            jobs[_jobId].creationTime + 90 days < block.timestamp,
+            "90 days are not completed yet"
+        );
+        uint256 bounty = jobs[_jobId].bounty;
+        uint256 referralId = jobs[_jobId].hiredReferralId;
+        bytes32 candidateEmail = referrals[referralId].candidateEmail;
+        address referrer = referrals[referralId].referrer;
+        address candidate = candidates[candidateEmail].candidateAddress;
+        balances[msg.sender] -= bounty;
+        balances[referrer] += (bounty * 6500) / 10_000;
+        balances[candidate] += (bounty * 1000) / 10_000;
+        balances[frontDoorAddress] = (bounty * 2500) / 10_000;
+        jobs[_jobId].isDisbursed = true;
+        emit BountyDisbursed(_jobId);
+    }
+
+    /// Claim rewards using Pull over Push pattern
+    function claimRewards() external nonReentrant {
```

Comment:
is it worth looking at: https://docs.openzeppelin.com/contracts/4.x/api/security#PullPayment

Reply:
despite that I'm using everything from OZ, I'm not sure if using their implementation will add value. 
It's actually very similar to the implementation.